pkg/fuzzy/aws_translate.go

```diff
@@ -0,0 +1,178 @@
+	client awsClient
+}
+
+type AWSTranslateOption func(*AWSTranslate) error
+
+// WithAWSClient sets the AWS Translate client.
+func WithAWSClient(awsc awsClient) AWSTranslateOption {
+	return func(awst *AWSTranslate) error {
+		awst.client = awsc
+		return nil
+	}
+}
+
+// WithDefaultAWSClient creates a new AWS Translate client with credentials from the viper.
+func WithDefaultAWSClient(ctx context.Context) AWSTranslateOption {
+	return func(awst *AWSTranslate) error {
+		accessKey := viper.GetString("other.aws_translate.access_key")
+		if accessKey == "" {
+			return fmt.Errorf("with default client: AWS translate access key is not set")
+		}
+
+		secretKey := viper.GetString("other.aws_translate.secret_key")
+		if secretKey == "" {
+			return fmt.Errorf("with default client: AWS translate secret key is not set")
+		}
+
+		region := viper.GetString("other.aws_translate.region")
+		if region == "" {
+			return fmt.Errorf("with default client: AWS translate region is not set")
+		}
+
+		// Create a new AWS SDK config
+		cfg, err := config.LoadDefaultConfig(ctx,
+			config.WithRegion(region),
+			config.WithHTTPClient(&http.Client{Transport: otelhttp.NewTransport(http.DefaultTransport)}),
+			config.WithCredentialsProvider(credentials.NewStaticCredentialsProvider(accessKey, secretKey, "")),
+		)
+		if err != nil {
+			return fmt.Errorf("failed to load default AWS SDK configuration: %w", err)
+		}
+
+		awst.client = translate.NewFromConfig(cfg)
+
+		return nil
+	}
+}
+
+// NewAWSTranslate creates a new AWS Translate service.
+func NewAWSTranslate(ctx context.Context, opts ...AWSTranslateOption) (*AWSTranslate, error) {
+	awst := &AWSTranslate{}
+
+	for _, opt := range opts {
+		if optErr := opt(awst); optErr != nil {
+			return nil, fmt.Errorf("apply opt: %w", optErr)
+		}
+	}
+
+	// Ping the AWS Translate API to ensure that the client is working.
+	_, err := awst.client.TranslateText(ctx, &translate.TranslateTextInput{
+		SourceLanguageCode: ptr("en"),
+		TargetLanguageCode: ptr("lv"),
+		Text:               ptr("Hello World!"),
+	})
+	if err != nil {
+		return nil, fmt.Errorf("AWS translate client: ping AWS translate: %w", err)
+	}
+
+	return awst, nil
+}
+
+// --------------------Methods--------------------
+
+func (a *AWSTranslate) Translate(ctx context.Context, messages *model.Messages) (*model.Messages, error) {
+	if messages == nil {
+		return nil, nil
+	}
+
+	if len(messages.Messages) == 0 {
+		return &model.Messages{Language: messages.Language, Original: messages.Original}, nil
+	}
+
+	// skip locale part if region is not a country,
+	// AWS only supports ISO 3166 2-digit country codes.
+	// https://docs.aws.amazon.com/translate/latest/dg/what-is-languages.html
+	targetLanguage := messages.Language.String()
+
+	if region, _ := messages.Language.Region(); !region.IsCountry() {
+		baseLang, _ := messages.Language.Base()
+		targetLanguage = baseLang.String()
+	}
+
+	translatedTexts := make([]string, 0, len(messages.Messages))
```

Comment:
No need to use `translatedTexts`, declare `translatedMessages` here and append translated texts to it when you receive a translation from AWS Translate API.